pkg/kubectl/cmd/cmd_test.go

```diff
@@ -209,7 +213,12 @@ type testFactory struct {
 	Err          error
 }
 
-func NewTestFactory() (*cmdutil.Factory, *testFactory, runtime.Codec, runtime.NegotiatedSerializer) {
+type fakeFactory struct {
+	tf    *testFactory
```

Comment:
Would it make sense to have a general fakeFactory for tests, that will have all the methods implemented, with empty bodies, of course. Then in commands you'd be embedding it and just providing an implementation to whichever method you need. This should simplify testing, imho.